main.tf

```diff
@@ -51,6 +51,7 @@ resource "aws_lambda_function" "main_s3" {
   s3_bucket         = "${var.s3_bucket}"
   s3_key            = "${var.s3_key}"
   s3_object_version = "${var.s3_object_version}"
+  source_code_hash = "${base64sha256(file(var.s3_key))}"
```

Comment:
I suspect this won't work since the `file()` function actually reads the content of a file to a string. And since `s3_key` is just a path on S3, there is nothing to read locally.

Reply:
Should probably be looking at `s3_object_version`.